examples/applications/plot_time_series_lagged_features.py

```diff
@@ -19,28 +19,44 @@
 # Analyzing the Bike Sharing Demand dataset
 # -----------------------------------------
 #
-# We start by loading the data from the OpenML repository.
+# We start by loading the data from the OpenML repository
+# as a pandas dataframe. This will be replaced with Polars
+# after `fetch_openml` will add a native support for it.
+# We convert to Polars for feature engineering, as it presents several advantages
+# here:
+#
+# - It parallelises the expressions within the `with_columns` statement.
+# - It automatically caches common subexpressions which are reused in multiple
+#   expressions (like `pl.col("count").shift(1)` below). See
+#   https://docs.pola.rs/user-guide/lazy/optimizations/ for more information.
+# - It is very memory-efficient.
+# - It is stricter about data types and schemas, allowing us to catch errors earlier
+#   in our development workflow.

```

Comment:
I don't think that this our place here to emphasize why one should use polars. I think that the example should only concentrate on the "consumer" part of a scikit-learn user meaning, that the underlying model will deal with a polars dataframe.

I think here the note regarding the caveats for `fetch_openml` is enough.

Reply:
Hey 👋 

I just wanted to flag then when I opened https://github.com/scikit-learn/scikit-learn/pull/28576, one of the comments was

> maybe we can highlight the benefit with a comment in the code or as narrative text

So, just looping in @ArturoAmorQ to make sure that scikit-learn devs are aligned here about highlighting benefits in examples, and that you're on the same page about the direction this is going in 🤗

Reply:
I agree with @glemaitre that comments regarding the benefits should remain as minimalistic as possible to keep the focus on the example's main narrative. Maybe in this case in particular the most relevant reason is the caching of common subexpressions?

Reply:
I left only the caching of common subexpressions and removed other reasons for using Polars.